File: C++/C++Basic/04.Conditional-Statements-Exercise/08.lunchBreak.cpp
#include <iostream>
#include <string>
#include <cmath>


using namespace std;

int main()
{
    string seriesName;
    int episodeDuration, breakDuration;

    getline(cin, seriesName);
    cin >> episodeDuration >> breakDuration;

    double lunchTime = breakDuration * 0.125;
    double restTime = breakDuration * 0.25;

    double totalTime = episodeDuration + lunchTime +restTime;

    if(totalTime <= breakDuration){
        cout << "You have enough time to watch " << seriesName << " and left with " << ceil(breakDuration - totalTime) << " minutes free time." << endl;
    } 
    else{
       cout << "You don't have enough time to watch " <<seriesName << ", you need "<< ceil(totalTime - breakDuration) << " more minutes." << endl;
    }

    return 0;

}

File: Python/Basic/04.Conditional-Statements-Exercise/08.lunch_break.py
import math

film = input()
episode_duration = int(input())
rest_duration = int(input())

time_for_lunch = rest_duration / 8
time_for_rest = rest_duration / 4

total_time = rest_duration - time_for_lunch - time_for_rest

if total_time >= episode_duration:
    diff = math.ceil(total_time - episode_duration)
    print(f"You have enough time to watch {film} and left with {diff} minutes free time.")
else:
    timeNeeded = math.ceil(episode_duration - total_time)
    print(f"You don't have enough time to watch {film}, you need {timeNeeded} more minutes.")